### DISCUSSION TITLE
What database I can use as data source of featurebyte?

### DISCUSSION THREAD

**Discussion by wanalytics8** [featurebyte/featurebyte#1473] [Q&A] [answered]
Hi, by reading the tutorials and documents, I know featurebyte can "connect to data warehouse" that is : Snowflake, Spark, Databricks

My question is, can I use say Big Query or Redshift as source data of the feature? If yes then where do I change the config?

Thanks!

**Comment by colinpriest** [comment]
Hi Wanalytics8,

You are correct that the FeatureByte library currently supports Snowflake, Spark, and Databricks. It's on our roadmap to expand support to Big Query due to its popularity in enterprises. Which data warehouses are you using at your workplace?

Colin

**Reply by wanalytics8** [reply]
Hi 

Thanks for the swift reply, we use Google BQ and Redshift.. so got to wait :)

Looking at https://docs.featurebyte.com/0.3/get_started/connect_to/spark/  ,as long as I have a thrift server, and credential for storage service, then I can connect?

PySpark maybe possible?
```
from pyspark.sql import SparkSession
from pyspark.sql.functions import flatten, udf, col
from pyspark.sql.types import BooleanType
bucket = 'my_bucket'
spark = SparkSession.builder\
    .appName("pyspark-test")\
    .config("spark.jars", "gs://spark-lib/bigquery/spark-bigquery-latest.jar")\
    .getOrCreate()

df = spark.read \
    .format("bigquery") \
    .load('bigquery-public-data.chicago_taxi_trips.taxi_trips').cache()
```

**Reply by kchua78** [reply, marked as answer]
Hi Wanalytics8,

Featurebyte can work with any table that is accessible from the thrift service.

You can register the BigQuery table in your spark catalog using PySpark or Spark SQL to make it available, e.g.:
```
CREATE TABLE taxi_data.taxi_trips using bigquery OPTIONS(table="bigquery-public-data.chicago_taxi_trips.taxi_trips");
```

Then it can be used like any other source table
```
taxi_trips = data_source.get_source_table(
    database_name="spark_catalog",
    schema_name="taxi_data", 
    table_name="taxi_trips"
).create_event_table(
    name="Taxi Trips",
    event_timestamp_column="trip_end_timestamp",
    event_id_column="unique_key",
    ...
)
```

**Reply by wanalytics8** [reply]
Thanks a lot.
